data_model/src/events/data/filters.rs

```diff
@@ -1,192 +1,734 @@
+        Decode,
+        Encode,
         Deserialize,
+        Serialize,
         IntoSchema,
     )]
-    #[serde(bound(
-        deserialize = "<<T as HasOrigin>::Origin as Identifiable>::Id: Deserialize<'de>"
-    ))]
-    #[serde(transparent)]
-    #[repr(transparent)]
-    pub struct OriginFilter<T: HasOrigin>(pub(super) <T::Origin as Identifiable>::Id)
-    where
-        <T::Origin as Identifiable>::Id:
-            Debug + Clone + Eq + Ord + Decode + Encode + Serialize + IntoSchema;
+    pub struct ExecutorEventFilter {
+        // executor is a global entity, so no id here
+        /// Matches only event from this set
+        pub(super) event_set: ExecutorEventSet,
+    }
+}
+
+impl PeerEventFilter {
+    /// Creates a new [`PeerEventFilter`] accepting all [`PeerEvent`]s.
+    pub const fn new() -> Self {
+        Self {
+            id_matcher: None,
+            event_set: PeerEventSet::all(),
+        }
+    }
+
+    /// Modifies a [`PeerEventFilter`] to accept only [`PeerEvent`]s originating from ids matching `id_matcher`.
+    #[must_use]
+    pub fn only_from(mut self, id_matcher: PeerId) -> Self {
+        self.id_matcher = Some(id_matcher);
+        self
+    }
+
+    /// Modifies a [`PeerEventFilter`] to accept only [`PeerEvent`]s of types contained in `event_set`.
+    #[must_use]
+    pub const fn only_events(mut self, event_set: PeerEventSet) -> Self {
+        self.event_set = event_set;
+        self
+    }
+}
+
+impl Default for PeerEventFilter {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+#[cfg(feature = "transparent_api")]
+impl EventFilter for PeerEventFilter {
+    type Event = super::PeerEvent;
+
+    fn matches(&self, event: &Self::Event) -> bool {
+        if let Some(id_matcher) = &self.id_matcher {
+            if id_matcher != event.origin_id() {
+                return false;
+            }
+        }
+
+        if !self.event_set.matches(event) {
+            return false;
+        }
+
+        true
+    }
 }
 
-mod accept_all_as_string {
-    //! Module to (de-)serialize `FilterOpt::AcceptAll` variant as string
+impl DomainEventFilter {
+    /// Creates a new [`DomainEventFilter`] accepting all [`DomainEvent`]s.
+    pub const fn new() -> Self {
+        Self {
+            id_matcher: None,
+            event_set: DomainEventSet::all(),
+        }
+    }
+
+    /// Modifies a [`DomainEventFilter`] to accept only [`DomainEvent`]s originating from ids matching `id_matcher`.
+    #[must_use]
+    pub fn only_from(mut self, id_matcher: DomainId) -> Self {
+        self.id_matcher = Some(id_matcher);
+        self
+    }
+
+    /// Modifies a [`DomainEventFilter`] to accept only [`DomainEvent`]s of types contained in `event_set`.
+    #[must_use]
+    pub const fn only_events(mut self, event_set: DomainEventSet) -> Self {
+        self.event_set = event_set;
+        self
+    }
+}
+
+impl Default for DomainEventFilter {
+    fn default() -> Self {
+        Self::new()
+    }
+}
```

Comment:
could this not be derived? Wouldn't it be a good default for `EventSet` to match all all

Reply:
I am not comfortable to make a default for event _set_ a full set. For sets in rust (and in general) the default is the empty set, so I think it would be confusing to break this expectation in a `Default` impl. We don't have a use for the `Default` impl making an empty set now, so I opted to just not include it to avoid confusion.